### PR TITLE
Feat: Cloudinary integration

### PR DESCRIPTION
**Pull request by kirkas**
### Security 

This should address the following security risks:
- [BASE-10: Malicious SVGs](https://jira.coinbase-corp.com/projects/BASE/issues/BASE-10?filter=allopenissues)
- [BASESEC-64: Load SVG via proxy](https://jira.coinbase-corp.com/browse/BASESEC-64)
- [SPA-9999: Design review](https://jira.coinbase-corp.com/browse/SPA-9999)

### Integration

All images that are not hosted by us will automatically be proxied via `ImageCloudinary`.

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/web/src/components/ImageCloudinary/index.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+  const cloudinaryFetchUrl = getCloudinaryMediaUrl({
+    media: absoluteSrc,
+    width: Number(width),
+  });
+
+  const shouldUploadToCloudinary = isDataUrl(absoluteSrc) || absoluteSrc.length > 255;
+
+  useEffect(() => {
+    // Some image needs to be upload befored being proxied
```
Comment (JFrankfurt) [inline apps/web/src/components/ImageCloudinary/index.tsx:45]:
```suggestion
    // Some images need to be uploaded before being proxied
```

File: apps/web/src/components/ImageCloudinary/index.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    // dataUrl & long Urls (dataUrl in disguise) needs to be uploaded to Cloudinary
```
Comment (JFrankfurt) [inline apps/web/src/components/ImageCloudinary/index.tsx:46]:
```suggestion
    // dataUrl & long Urls need to be uploaded to Cloudinary
```

File: apps/web/app/api/cloudinaryUrl/route.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    const response = await cloudinary.api.resources_by_ids([`${folderName}/${assetId}`]);
+    if (response && response.resources.length > 0) {
+      const image = response.resources[0];
+      return image.secure_url;
+    } else {
+      return false;
+    }
+  } catch (error) {
+    // For other errors, log and assume the asset doesn't exist
+    console.error('Error checking if asset exists in Cloudinary:', error);
```
Comment (JFrankfurt) [inline apps/web/app/api/cloudinaryUrl/route.ts:38]: do we want an external log of this?

File: apps/web/app/api/cloudinaryUrl/route.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      folder: folderName,
+      format: 'webp',
+      transformation: {
+        width: width,
+      },
+    });
+
+    return result;
+  } catch (error) {
+    console.log('Failed to upload asset', error);
```
Comment (JFrankfurt) [inline apps/web/app/api/cloudinaryUrl/route.ts:59]: .error or remove

**Review by JFrankfurt — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for JFrankfurt @ 2024-10-15 15:54:30 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by JFrankfurt — APPROVED** [review]

**Comment by JFrankfurt** [comment]
🔥

**Pull request merged**